app/src/main/java/com/example/vybe/MyVibesAdapter.java

```diff
@@ -45,30 +50,40 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
                     .inflate(this.resource, parent, false);
         }
 
-        TextView dateField = view.findViewById(R.id.view_date_text_view);
-        TextView reasonField = view.findViewById(R.id.my_reason_text_view);
-        ImageView vibeImage = view.findViewById(R.id.imageView);
+        dateField = view.findViewById(R.id.view_date_text_view);
+        reasonField = view.findViewById(R.id.my_reason_text_view);
+        vibeImage = view.findViewById(R.id.image_view);
 
         VibeEvent vibeEvent = vibeEventList.get(position);
+        populateVibeAdapterFields(vibeEvent);
 
+        return view;
+    }
+
+    /**
+     * This will populate the appropriate fields for displaying Vibe
+     * Event details on the VibesAdapter view
+     * @param vibeEvent
+     *      The Vibe Event object containing details to be displayed
+     */
+    // TODO:
+    // this can be moved to a controller class and could potentially be used in common with
+    // the ViewVibeActivity method: populateVibeEventDetails
+    public void populateVibeAdapterFields(VibeEvent vibeEvent) {
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern(" d, YYYY h:mm a", Locale.ENGLISH);
         LocalDateTime dateTime = vibeEvent.getDateTime();
         String month = dateTime.getMonth().getDisplayName(TextStyle.FULL_STANDALONE, Locale.ENGLISH);
+        String dateFieldText = month + dateTime.format(formatter);
+        dateField.setText(dateFieldText);
+        vibeImage.setImageResource(vibeEvent.getVibe().getEmoticon());
 
-        dateField.setText(month + dateTime.format(formatter));
-
-        if (vibeEvent.getVibe() != null) {
-            vibeImage.setImageResource(vibeEvent.getVibe().getEmoticon());
-        } else {
-            vibeImage.setImageResource(R.drawable.ic_no_vibe);
-        }
-
+        // TODO: DO WE WANT TO DISPLAY REASON?
```

Comment:
my answer would be no, i was going to suggest to remove it since we were having issues with displaying them anyway